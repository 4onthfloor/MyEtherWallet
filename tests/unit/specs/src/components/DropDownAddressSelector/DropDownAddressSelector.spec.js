import DropDownAddressSelector from '@/components/DropDownAddressSelector/DropDownAddressSelector.vue';
import { shallowMount } from '@vue/test-utils';

import { Tooling } from '@@/helpers';

describe('DropDownAddressSelector.vue', () => {
  let localVue, i18n, store, wrapper;

  beforeAll(() => {
    const baseSetup = Tooling.createLocalVueInstance();
    localVue = baseSetup.localVue;
    i18n = baseSetup.i18n;
    store = baseSetup.store;
  });

  beforeEach(() => {
    wrapper = shallowMount(DropDownAddressSelector, {
      localVue,
      i18n,
      store,
      attachToDocument: true,
      computed: {
        addresses() {
          return [
            {
              address: 'mewtopiax.eth',
              currency: 'ETH',
              nickname: 1
            },
            {
              address: '1DECAF2uSpFTP4L1fAHR8GCLrPqdwdLse9',
              currency: 'ETH',
              nickname: 2
            }
          ];
        }
      }
    });
  });

  afterAll(() => {
    wrapper.destroy();
  });

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('render correct addresses', async () => {
    const dropdownOpen = wrapper.find('.dropdown-open-button');
    dropdownOpen.trigger('click');
    await wrapper.vm.$nextTick();
    expect(wrapper.vm.$data.dropdownOpen).toBe(true);
    const addressElements = wrapper.findAll(
      '.dropdown-list-box .listed-address'
    );
    addressElements.wrappers.forEach((addressElement, i) => {
      expect(addressElement.text().trim()).toContain(
        wrapper.vm.addresses[i].address
      );
    });
  });

  it('should dismount properly', () => {
    expect(wrapper.destroy()).toBe(undefined);
  });
});
