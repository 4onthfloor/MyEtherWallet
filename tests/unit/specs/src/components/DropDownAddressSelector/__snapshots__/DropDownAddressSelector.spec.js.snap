// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropDownAddressSelector.vue renders correctly 1`] = `
<div class="drop-down-address-selector">
  <div class="dropdown--title">
    <h4></h4> <button class="title-button prevent-user-select">
      Copy
    </button>
  </div>
  <div class="dropdown--content">
    <div class="dropdown-input-box">
      <div><input placeholder="Please enter the address" type="text" autocomplete="off"></div> <span class="save-addr-txt disabled-txt">Save Address</span>
      <div class="blockie-place-holder-image"></div>
      <!---->
      <div class="dropdown-open-button"><i aria-hidden="true" class="fa fa-chevron-down"></i>
        <!---->
      </div>
    </div>
    <!---->
  </div>
  <address-book-modal-stub modalaction="add" title="interface.address-book.add-new" currentidx="0" selectedaddress=""></address-book-modal-stub>
</div>
`;
